app.js

```diff
@@ -1,90 +1,156 @@
-                    }
 
-                    connection.query(`INSERT INTO ${tableName} VALUES (${rowToInsert.map(x => {
-                        x = x.replace(/"/g, '\\"');
-                        return '"' + x + '"';
-                    }).join()})`, error => {
-                        // Close the connection once all queries are complete
-                        if (++completedQueryCount === array.length) {
-                            connection.end();
+                if (insertingCase !== "--append") {
+                    if (colsName.join().indexOf(" ") > -1 || colsName.join().indexOf("'") > -1 || colsName.join().indexOf('"') > -1) {
+                        for (let column of colsName) {
+
+                            if (column.indexOf(" ") > -1 || column.indexOf("'") > -1 || column.indexOf('"') > -1) {
+                                createTableQuery += '`' + column + '`' + " TEXT, ";
+                            } else {
+                                createTableQuery += column + " TEXT, ";
+                            }
                         }
+                        createTableQuery = createTableQuery.slice(0, -2) + ")";
+                    } else {
+                        createTableQuery += colsName.join(" TEXT, ") + " TEXT)";
+                    }
+                }
+                if (insertingCase === "--overwrite") {
 
+                    connection.query(`DROP TABLE IF EXISTS ${tableName.replace(/'/g, "\\'")}`, error => {
                         if (error) {
                             console.error(error);
                             return;
                         }
+                        CreateTable(createTableQuery, data, tableName, insertRows, completedQueryCount, connection);
                     });
-                });
+
+                } else if (insertingCase === "--append") {
+
+                    insertRows(tableName, data, completedQueryCount, connection);
+
+                } else {
+
+                    CreateTable(createTableQuery, data, tableName, insertRows, completedQueryCount, connection);
+
+                }
             });
         });
     });
-});
+}
+
+
+let args = process.argv;
+filesToInsert = [];
+for (let i = 2; i < args.length; i++) {
+    if (args[i] === "--files") {
+        isUserSpecifiedFilesToInsert = true;
+    }
+    if (args[i] === '--append' || args[i] === '--overwrite') {
+        insertingCase = args[i];
+    }
+    if (args[i].indexOf('.csv') > -1 && isUserSpecifiedFilesToInsert) {
+        filesToInsert.push(args[i]);
+    }
+}
+
+
+
+if (isUserSpecifiedFilesToInsert) {
+    // in case if someone did : [node app.js --files] and runs the program
+    if (filesToInsert.length) {
+        readAndParseAndInsert(filesToInsert, "./", insertingCase, CreateTable, insertRows);
+    } else {
+        console.error(new Error('No CSV files specified'));
+    }
+} else {
+    fs.readdir('./files/', 'utf8', (error, files) => {
```

Comment:
Why do you specify utf8 here? According to the documentation, it's completely optional:
https://nodejs.org/api/fs.html#fs_fs_readdir_path_options_callback
So I'm curious to know why you've explicitly specified utf8.